azkaban-common/src/test/java/azkaban/project/JdbcProjectLoaderTest.java

```diff
@@ -214,6 +213,96 @@ private static void clearDB() {
+        if (!isTestSetup()) {
+            return;
+        }
+
+        ProjectLoader loader = createLoader();
+        String projectName = "mytestProject";
+        String projectDescription = "This is my new project";
+        User user = new User("testUser");
+
+        Project project =
+            loader.createNewProject(projectName, projectDescription, user);
+
+        Permission perm = new Permission(0x2);
+        loader.updatePermission(project, user.getUserId(), perm, false);
+        loader.updatePermission(project, "group", perm, true);
+
+        Permission permOverride = new Permission(0x6);
+        loader.updatePermission(project, user.getUserId(), permOverride, false);
+
+        Project fetchedProject = loader.fetchProjectByName(project.getName());
+        assertProjectMemberEquals(project, fetchedProject);
+        Assert.assertEquals(permOverride,
+            fetchedProject.getUserPermission(user.getUserId()));
+    }
+
+    /** Default Test case for fetchProjectByName **/
+    @Test
+    public void testProjectRetrivalByFetchProjectByName()
+        throws ProjectManagerException {
+        if (!isTestSetup()) {
+            return;
+        }
+
+        ProjectLoader loader = createLoader();
+        String projectName = "mytestProject";
+        String projectDescription = "This is my new project";
+        User user = new User("testUser");
+
+        Project project =
+            loader.createNewProject(projectName, projectDescription, user);
+
+        Project fetchedProject = loader.fetchProjectByName(project.getName());
+        assertProjectMemberEquals(project, fetchedProject);
+    }
+
+    /** Default Test case for fetchProjectByName **/
+    @Test
+    public void testDuplicateRetrivalByFetchProjectByName()
+        throws ProjectManagerException {
+        if (!isTestSetup()) {
+            return;
+        }
+
+        ProjectLoader loader = createLoader();
+        String projectName = "mytestProject";
+        String projectDescription = "This is my new project";
+        User user = new User("testUser");
+
+        Project project =
+            loader.createNewProject(projectName, projectDescription, user);
+
+        loader.removeProject(project, user.getUserId());
+
+        Project newProject =
+            loader.createNewProject(projectName, projectDescription, user);
+
+        Project fetchedProject = loader.fetchProjectByName(project.getName());
+        Assert.assertEquals(newProject.getId(), fetchedProject.getId());
+
+    }
+
+    /** Test case for NonExistantProject project fetch **/
+    @Test
+    public void testInvalidProjectByFetchProjectByName() {
+        if (!isTestSetup()) {
+            return;
+        }
+        ProjectLoader loader = createLoader();
+        try {
+            loader.fetchProjectByName("NonExistantProject");
+        } catch (ProjectManagerException ex) {
+            System.out.println("Test true");
+        }
+        Assert.fail("No exception");
```

Comment:
Maybe make this message more clear i.e "Expecting exception, but didn't get one"